internal/command/views/test.go

```diff
@@ -732,6 +736,276 @@ func (t *TestJSON) TFCRetryHook(attemptNum int, resp *http.Response) {
+	xmlSrc, err := junitXMLTestReport(suite)
+	if err != nil {
+		v.err = err
+		return
+	}
+	err = os.WriteFile(v.filename, xmlSrc, 0660)
+	if err != nil {
+		v.err = err
+		return
+	}
+}
+
+func (v *TestJUnitXMLFile) File(file *moduletest.File, progress moduletest.Progress) {}
+
+func (v *TestJUnitXMLFile) Run(run *moduletest.Run, file *moduletest.File, progress moduletest.Progress, elapsed int64) {
+}
+
+func (v *TestJUnitXMLFile) DestroySummary(diags tfdiags.Diagnostics, run *moduletest.Run, file *moduletest.File, state *states.State) {
+}
+
+func (v *TestJUnitXMLFile) Diagnostics(run *moduletest.Run, file *moduletest.File, diags tfdiags.Diagnostics) {
+}
+
+func (v *TestJUnitXMLFile) Interrupted() {}
+
+func (v *TestJUnitXMLFile) FatalInterrupt() {}
+
+func (v *TestJUnitXMLFile) FatalInterruptSummary(run *moduletest.Run, file *moduletest.File, states map[*moduletest.Run]*states.State, created []*plans.ResourceInstanceChangeSrc) {
+}
+
+func (v *TestJUnitXMLFile) TFCStatusUpdate(status tfe.TestRunStatus, elapsed time.Duration) {}
+
+func (v *TestJUnitXMLFile) TFCRetryHook(attemptNum int, resp *http.Response) {}
+
+func junitXMLTestReport(suite *moduletest.Suite) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := xml.NewEncoder(&buf)
+	enc.EncodeToken(xml.ProcInst{
+		Target: "xml",
+		Inst:   []byte(`version="1.0" encoding="UTF-8`),
+	})
+	enc.Indent("", "  ")
+
+	// Some common element/attribute names we'll use repeatedly below.
+	suitesName := xml.Name{Local: "testsuites"}
+	suiteName := xml.Name{Local: "testsuite"}
+	caseName := xml.Name{Local: "testcase"}
+	nameName := xml.Name{Local: "name"}
+	testsName := xml.Name{Local: "tests"}
+	failuresName := xml.Name{Local: "failures"}
+	errorsName := xml.Name{Local: "errors"}
+
+	enc.EncodeToken(xml.StartElement{Name: suitesName})
+	for _, file := range suite.Files {
+		// Each test file is modelled as a "test suite".
+
+		// First we'll count the number of tests and number of failures/errors
+		// for the suite-level summary.
+		totalTests := len(file.Runs)
+		totalFails := 0
+		totalErrs := 0
+		for _, run := range file.Runs {
+			switch run.Status {
+			case moduletest.Fail:
+				totalFails++
+			case moduletest.Error:
+				totalErrs++
+			}
+		}
+		enc.EncodeToken(xml.StartElement{
+			Name: suiteName,
+			Attr: []xml.Attr{
+				{Name: nameName, Value: file.Name},
+				{Name: testsName, Value: strconv.Itoa(totalTests)},
+				{Name: failuresName, Value: strconv.Itoa(totalFails)},
+				{Name: errorsName, Value: strconv.Itoa(totalErrs)},
+			},
+		})
+
+		for _, run := range file.Runs {
+			// Each run is a "test case".
+
+			type WithMessage struct {
+				Message string `xml:"message,attr,omitempty"`
+				Body    string `xml:",cdata"`
+			}
+			type TestCase struct {
+				Name    string       `xml:"name,attr"`
+				Failure *WithMessage `xml:"failure,omitempty"`
+				Error   *WithMessage `xml:"error,omitempty"`
+				Stderr  *WithMessage `xml:"system-err,omitempty"`
+			}
```

Comment:
We should support `Skipped` in here as well

Reply:
I've pushed a new commit that handles `moduletest.Skip` both as a testcase element and as a summary attribute in the `testsuite` element.